x-pack/plugins/observability/server/services/slo/summary_client.ts

```diff
@@ -16,75 +16,96 @@ import {
 } from '@kbn/slo-schema';
 import moment from 'moment';
 import { SLO_DESTINATION_INDEX_PATTERN } from '../../assets/constants';
-import { DateRange, SLO, SLOId, Summary } from '../../domain/models';
+import { DateRange, SLO, Summary } from '../../domain/models';
 import { computeSLI, computeSummaryStatus, toErrorBudget } from '../../domain/services';
 import { toDateRange } from '../../domain/services/date_range';
 
+// TODO: Change name of this service...
+// It does compute a summary but from the rollup data.
 export interface SummaryClient {
-  fetchSummary(sloList: SLO[]): Promise<Record<SLOId, Summary>>;
+  computeSummary(slo: SLO, instanceId?: string): Promise<Summary>;
 }
 
 export class DefaultSummaryClient implements SummaryClient {
   constructor(private esClient: ElasticsearchClient) {}
 
-  async fetchSummary(sloList: SLO[]): Promise<Record<SLOId, Summary>> {
-    const dateRangeBySlo = sloList.reduce<Record<SLOId, DateRange>>((acc, slo) => {
-      acc[slo.id] = toDateRange(slo.timeWindow);
-      return acc;
-    }, {});
-    const searches = sloList.flatMap((slo) => [
-      { index: SLO_DESTINATION_INDEX_PATTERN },
-      generateSearchQuery(slo, dateRangeBySlo[slo.id]),
-    ]);
+  async computeSummary(slo: SLO, instanceId: string = ALL_VALUE): Promise<Summary> {
```

Comment:
💬 Simplification: Only one SLO and an optional instanceId parameters.
The query is simpler, and if the slo has a groupBy key defined, and we asked for a specific `instanceId` value, we add an extra filter on it.